File: machine-v3/internal/model/event.go
package model

import "time"

// Event represents an event in the system
type Event struct {
	CaseID    string                 `json:"case_id"`
	Timestamp time.Time              `json:"timestamp"`
	EventType string                 `json:"event_type"`
	Data      map[string]interface{} `json:"data"`
}

File: web/engines/models/event.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum
from typing import Any


@dataclass
class EventType(str, Enum):
    SUBMITTED = "SUBMITTED"
    DECIDED = "DECIDED"
    IN_REVIEW = "IN_REVIEW"
    OBJECTED = "OBJECTED"


@dataclass
class Event:
    """Event"""

    event_type: EventType
    timestamp: datetime
    data: dict[str, Any]
